src/test/java/com/wikia/webdriver/common/core/imageutilities/ImageGenerator.java

```diff
@@ -0,0 +1,78 @@
+package com.wikia.webdriver.common.core.imageutilities;
+
+import com.wikia.webdriver.common.contentpatterns.PageContent;
+import com.wikia.webdriver.common.core.CommonUtils;
+import com.wikia.webdriver.common.logging.PageObjectLogging;
+
+import java.awt.*;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+
+import javax.imageio.ImageIO;
+
+/**
+ * @ownership: Quality Team
+ */
+public class ImageGenerator {
+
+  private String imagePath;
+  private String imageFolder = PageContent.IMAGE_UPLOAD_RESOURCES_PATH;
+  BufferedImage imageBuffer;
+  private int imageWidth = 200;
+  private int imageHeight = 200;
+
+  public ImageGenerator() {
+    imageBuffer =
+        new BufferedImage(imageWidth, imageHeight,
+                          BufferedImage.TYPE_INT_ARGB);
+  }
+
+  /**
+   * Generates unique 200x200(px) .png image, with random text in its center
+   */
+  public void generateImageWithRandomText() {
+
+    String imageExtension = "png";
+    imagePath = imageFolder + "random_image." + imageExtension;
+    String fontName = "TimesRoman";
+    int fontStyle = Font.BOLD;
+    int fontSize = 20;
+    Color fontColor = Color.BLUE;
+    int textLength = 16;
+    String imageText = getRandomText(textLength);
+
+    String actionName = "generate random image";
+    String actionDescription = "generated image with random text: " + imageText;
+
+    Graphics2D g2 = imageBuffer.createGraphics();
+    g2.setPaint(fontColor);
+    Font font = new Font(fontName, fontStyle, fontSize);
+    g2.setFont(font);
+    FontMetrics fontMetrics = g2.getFontMetrics();
+    int stringWidth = fontMetrics.stringWidth(imageText);
+    int stringHeight = fontMetrics.getAscent();
+
+    // Draw the text in the middle of the image
+    g2.drawString(imageText, (imageWidth - stringWidth) / 2, imageHeight / 2);
+
+    try {
+      if (ImageIO.write(imageBuffer, imageExtension, new File(imagePath))) {
+        PageObjectLogging.logOnLowLevel(actionName, actionDescription, true);
+      }
+    } catch (IOException e) {
+      throw new RuntimeException("new image couldn't be saved");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)